### PR TITLE
fix: Resolve Fails on Windows

### LINKED ISSUE #1434: Resolve Artifacts Fails on Windows
**Describe the bug**
resolveArtifacts for windows fails due to the failure of privilege's that are required to create a symbolic link.
Files.kt
```
 Exception in thread "main" java.nio.file.FileSystemException:  A required privilege is not held by the client.
        at sun.nio.fs.WindowsException.translateToIOException(WindowsException.java:86)
        at sun.nio.fs.WindowsException.rethrowAsIOException(WindowsException.java:97)
        at sun.nio.fs.WindowsException.rethrowAsIOException(WindowsException.java:102)
        at sun.nio.fs.WindowsFileSystemProvider.createSymbolicLink(WindowsFileSystemProvider.java:585)
        at java.nio.file.Files.createSymbolicLink(Files.java:1043)
        at flank.common.FilesKt.createSymbolicLink(Files.kt:17)
```

**To Reproduce**
It was tested on two different windows 10 machines. Windows 10 Home(2004) & Pro(1903).
Both fail. 
Reproduce by running it on Windows 10 native. Not a virtual machine.

**Details (please complete the following information):**
Tested on commit - a7380ff0f3f2a71251593932daefdbbd53291917

Activating developer mode on Windows 10 Pro does not alleviate the problem.
We may need to elevate the privilege's of the java process or investigate a way around this issue.
Ideas.
1. Do an actual File copy for windows instead of a symbolic link
2. Checkout the conditions of this flag `SYMBOLIC_LINK_FLAG_ALLOW_UNPRIVILEGED_CREATE` and which OpenJDK supports this
3. Elevate the user process so it can complete successfully for windows. (Maybe execute a separate bat file or some other way.)

### PR DESCRIPTION
**Pull request by Sloox**
Fixes #1434 

Simple solution. Increases space used but works without process elevation.
works on Pro + Home.

### CODE REVIEW

**Comment by Sloox** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by Sloox** [comment]
recheck

**Comment by Sloox** [comment]
@flank-it

File: common/src/main/kotlin/flank/common/Files.kt [hunk 1]
```diff
@@ -14,6 +15,24 @@ val userHome: String by lazy {
...
+                Files.copy(source, destination)
+            } catch (e: IOException) {
+                logLn(e.localizedMessage)
+            }
+        }
+}
+
 fun createSymbolicLink(
     link: String,
     target: String
```
Comment (piotradamczyk5) [inline common/src/main/kotlin/flank/common/Files.kt:38]: Should not we apply similar logic for this method?

Comment (Sloox) [reply common/src/main/kotlin/flank/common/Files.kt:38]: We can but then its no longer creating a SymbolicLink and the name should change.

Comment (jan-gogo) [reply common/src/main/kotlin/flank/common/Files.kt:38]: Maybe this function signature should have a `typealias` so we could switch reference to function for specific linking strategy depending on OS type.

File: common/src/main/kotlin/flank/common/Files.kt [hunk 2]
```diff
@@ -14,6 +15,24 @@ val userHome: String by lazy {
...
+        .forEach { source: Path ->
+            val destination =
+                Paths.get(destinationDirectoryLocation, source.toString().substring(sourceDirectoryLocation.length))
+            try {
+                Files.copy(source, destination)
+            } catch (e: IOException) {
+                logLn(e.localizedMessage)
+            }
+        }
+}
```
Comment (piotradamczyk5) [inline common/src/main/kotlin/flank/common/Files.kt:34]: Could we add some os based tests to it to just make sure that it creates symbolic link?

Comment (Sloox) [reply common/src/main/kotlin/flank/common/Files.kt:39]: @piotradamczyk5 there is already test for symbolic linking done previously.

Comment (Sloox) [reply common/src/main/kotlin/flank/common/Files.kt:39]: its also fixed for windows
i will add tests for new functions added

Comment (Sloox) [reply common/src/main/kotlin/flank/common/Files.kt:37]: @piotradamczyk5 added

File: common/src/main/kotlin/flank/common/Files.kt (hunk 1 above)
Comment (Sloox) [reply common/src/main/kotlin/flank/common/Files.kt:53]: As mentioned we are copying for windows and symlinking for everyone else. I dont think there should be a typealias as its fairly self explanatory.

**Review by piotradamczyk5 — APPROVED** [review]

**Comment by adamfilipow92** [comment]
I try to test your solution but when I try to build Flank I get
```
Execution failed for task ':flank-scripts:checkIfVersionUpdated'.
> Flank scripts version is not updated, but files changed.
  Please update version according to schema: <breaking change>.<feature added>.<fix/minor change>

```

**Comment by pawelpasterz** [comment]
> I try to test your solution but when I try to build Flank I get
> 
> ```
> Execution failed for task ':flank-scripts:checkIfVersionUpdated'.
> > Flank scripts version is not updated, but files changed.
>   Please update version according to schema: <breaking change>.<feature added>.<fix/minor change>
> ```

Hey @adamfilipow92 I've implemented some changes, could you verify again?

**Comment by adamfilipow92** [comment]
> > I try to test your solution but when I try to build Flank I get
> > ```
> > Execution failed for task ':flank-scripts:checkIfVersionUpdated'.
> > > Flank scripts version is not updated, but files changed.
> >   Please update version according to schema: <breaking change>.<feature added>.<fix/minor change>
> > ```
> 
> Hey @adamfilipow92 I've implemented some changes, could you verify again?

Sure!

**Comment by adamfilipow92** [comment]
I think it's not related to this pr but two tests fail
```should create new epic with sub issues``` && ```should update existing epic``` from SDKUpdateTest, outputs looks the same.  Maybe we could skip these tests on windows and fix it in another ticket.

**Comment by pawelpasterz** [comment]
@adamfilipow92 thanks!
Fixed

**Comment by pawelpasterz** [comment]
@flank-it

**Comment by pawelpasterz** [comment]
@flank-it

File: .github/workflows/full_suite_integration_tests.yml [hunk 3]
```diff
@@ -161,6 +161,7 @@ jobs:
 
   trigger-pointer:
     runs-on: ubuntu-latest
+    if: github.ref == 'refs/heads/master'
```
Comment (piotradamczyk5) [inline .github/workflows/full_suite_integration_tests.yml:164]: why this kind of check?

**Review by piotradamczyk5 — CHANGES_REQUESTED** [review]

File: .github/workflows/full_suite_integration_tests.yml (hunk 3 above)
Comment (pawelpasterz) [reply .github/workflows/full_suite_integration_tests.yml:164]: Noticed during my work on this PR. Pointers should be only triggered by the master branch. We don't want comment trigger to change date.
If you want I can makanother PR for this change

Comment (piotradamczyk5) [reply .github/workflows/full_suite_integration_tests.yml:164]:
```suggestion
    if: ${{ github.event_name != 'issue_comment'}}
```

Comment (pawelpasterz) [reply .github/workflows/full_suite_integration_tests.yml:164]: Hm...same results, but sure, I can change it that way

Comment (piotradamczyk5) [reply .github/workflows/full_suite_integration_tests.yml]: same, but we do not depend on the branch name.

**Review by piotradamczyk5 — APPROVED** [review]

**Review by adamfilipow92 — APPROVED** [review]
- [X] Works on windows
- [X] It passes on windows

**Pull request merged**